org.OpenT2T.Sample.SuperPopular.Thermostat/Nest Thermostat/js/thingTranslator.js

```diff
@@ -0,0 +1,131 @@
+'use strict';
+var helper = require('opent2t-translator-helper-nest');
+var q = require('q');
+
+// logs device state
+function logDeviceState(device) {
+    if (typeof (device) !== 'undefined') {
+        console.log('  device.name          : ' + device.name);
+        console.log('  device.props         : ' + device.props);
+    } else {
+        console.log('device is undefined');
+    }
+};
+
+function validateArgumentType(arg, argName, expectedType) {
+    if (typeof arg === 'undefined') {
+        throw new Error('Missing argument: ' + argName + '. ' +
+            'Expected type: ' + expectedType + '.');
+    } else if (typeof arg !== expectedType) {
+        throw new Error('Invalid argument: ' + argName + '. ' +
+            'Expected type: ' + expectedType + ', got: ' + (typeof arg));
+    }
+}
+
+var device, props;
+
+// module exports, implementing the schema
+module.exports = {
+
+    initDevice : function(dev) {
+        device = dev;
+        console.log('Initializing device.');
+
+        device = dev;
+        validateArgumentType(device, 'device', 'object');
+        validateArgumentType(device.props, 'device.props', 'string');
+
+        props = JSON.parse(device.props);
+        validateArgumentType(props.access_token, 'device.props.access_token', 'string');
+        validateArgumentType(props.id, 'device.props.id', 'string');
+       
+        helper.init('thermostats',  props.id, props.access_token)
+        logDeviceState(device);
+	    console.log('Initialized.');
+    },
+
+    disconnect : function() {
+        console.log('disconnect called.');
+        logDeviceState(this.device);
+    },
+    
+    isTurnedOn : function() {
+        console.log('isTurnedOn called.');
+        return helper.getProperty('hvac_mode').then(state => {
+            console.log("state: "+ state);
+            return state != 'off';
+        });
+    },
+    
+    // Default to Heating. THe call must be followed by setMode(heating/cooling), as desired.
+    turnOn : function() {
+        console.log('turnOn called.');
+        return getMode().then( mode => {
+            if(mode == 'off')
+            {
+                return helper.setProperty({'hvac_mode' : 'heat-cool'});
+            }
+        });
```

Comment:
This function still needs to return a promise even when the mode is not off. So add an else block that returns a resolved promise.